uaclient/security_status.py

```diff
@@ -63,6 +64,18 @@ def get_origin_information_to_service_map():
     }
 
 
+@lru_cache(maxsize=None)
+def get_esm_cache():
+    try:
+        # If the rootdir folder doesn't contain any apt source info, the
+        # cache will be empty
+        cache = Cache(rootdir=ESM_APT_ROOTDIR)
+    except Exception:
```

Comment:
Well, considering the call in `try` it is improbable that we will introduce any bug.
If there is a problem with creating the instance (some bug in python3-apt?) we would have to catch a specific class instead (`apt.Error` iirc) and that would be kind of the same.
Besides that, given the advertisement-related nature of the feature, either we will catch the bug in our integration tests (if it is release related) or, if it is an individual instance, we can just consider the cache empty.

Reply:
> If there is a problem with creating the instance (some bug in python3-apt?) we would have to catch a specific class instead (apt.Error iirc) and that would be kind of the same.

Perhaps there's a misunderstanding here? I mean the inverse. If there is some unknown problem, it _should_ result in the exception not being caught. If you're testing specifically for "the rootdir folder doesn't contain any apt source info", then you should be able to quickly simulate that, determine what that exception is, and catch it specifically.

If you need eg. an apt hook to fail gracefully in the face of _any_ failure, then that should be done closer to the main function of the call stack.

Otherwise you can end up with trying to debug some edge case bug and being thwarted by catch-all exception handlers swallowing the real issue, combined with weird bug behaviour as a result of that.

Reply:
Oh no, but if the rootdir folder is empty, this does not result in an error at all. Nothing that we are doing here can result in any error and, if it does, we don't care. Besides that, security status isn't a part of the hook - it is a separate command which consumes this information. And, if something happens here, we don't really care OR it will be a general thing, caught by the integration tests as mentioned.